pmdarima/compat/sklearn.py

```diff
@@ -50,3 +51,46 @@ def safe_indexing(X, indices):
         return X[indices, :]
     # list or 1d array
     return X[indices]
+
+
+def _estimator_has(attr):
+    """Checks if the model has a given attribute.
+
+    Meant to be used along with `sklearn.utils.metaestimators.available_if`
+
+    Parameters
+    ----------
+    attr : str
+        The attribute to check the calling object for
+
+    Returns
+    -------
+    fn : callable
+        A function that will either raise an `AttributeError` if the attribute
+        does not exist, or True if it does.
+    """
+    def check(self):
+        # raise original `AttributeError` if `attr` does not exist
+        getattr(self, attr)
+        return True
+
+    return check
+
+
+def if_delegate_has_method(attr):
+    """Compat method to replace `sklearn.utils.metaestimators.if_delegate_has`
+
+    Older versions (<= 0.24.2) of sklearn support it, but newer versions use
+    `available_if` instead.
+
+    References
+    ----------
+    .. [1] https://git.io/JzKiv
+    .. [2] https://git.io/JzKiJ
+    """
+    if sklearn.__version__ <= "0.24.2":
```

Comment:
If they ever release a patch version `0.24.3` this will break. I don't think they will, but projects like Tensorflow will support patch versions on previous minor versions for several release cycles. I would just make this `if sklearn.__version__ < "1.0.0":`